gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/pooled_grpc_asyncio.py.j2

```diff
@@ -0,0 +1,445 @@
+                ignored if ``channel`` or ``ssl_channel_credentials`` is provided.
+            quota_project_id (Optional[str]): An optional project to use for billing
+                and quota.
+            client_info (google.api_core.gapic_v1.client_info.ClientInfo):
+                The client info used to send a user-agent string along with
+                API requests. If ``None``, then default info will be used.
+                Generally, you only need to set this if you're developing
+                your own client library.
+            always_use_jwt_access (Optional[bool]): Whether self signed JWT should
+                be used for service account credentials.
+
+        Raises:
+            google.auth.exceptions.MutualTlsChannelError: If mutual TLS transport
+              creation failed for any reason.
+            google.api_core.exceptions.DuplicateCredentialArgs: If both ``credentials``
+              and ``credentials_file`` are passed.
+            ValueError: if ``pool_size`` <= 0
+        """
+        if pool_size <= 0:
+            raise ValueError(f"invalid pool_size: {pool_size}")
+        self._ssl_channel_credentials = ssl_channel_credentials
+        self._stubs: Dict[Tuple[aio.Channel,str], Callable] = {}
+        self._next_idx = 0
+        {% if service.has_lro %}
+        self._operations_client: Optional[operations_v1.OperationsAsyncClient] = None
+        {% endif %}
+
+        if api_mtls_endpoint:
+            warnings.warn("api_mtls_endpoint is deprecated", DeprecationWarning)
+        if client_cert_source:
+            warnings.warn("client_cert_source is deprecated", DeprecationWarning)
+
+        if api_mtls_endpoint:
+            host = api_mtls_endpoint
+
+            # Create SSL credentials with client_cert_source or application
+            # default SSL credentials.
+            if client_cert_source:
+                cert, key = client_cert_source()
+                self._ssl_channel_credentials = grpc.ssl_channel_credentials(
+                    certificate_chain=cert, private_key=key
+                )
+            else:
+                self._ssl_channel_credentials = SslCredentials().ssl_credentials
+
+        else:
+            if client_cert_source_for_mtls and not ssl_channel_credentials:
+                cert, key = client_cert_source_for_mtls()
+                self._ssl_channel_credentials = grpc.ssl_channel_credentials(
+                    certificate_chain=cert, private_key=key
+                )
+
+        # The base transport sets the host, credentials and scopes
+        super().__init__(
+            host=host,
+            credentials=credentials,
+            credentials_file=credentials_file,
+            scopes=scopes,
+            quota_project_id=quota_project_id,
+            client_info=client_info,
+            always_use_jwt_access=always_use_jwt_access,
+            api_audience=api_audience,
+        )
+        self._quota_project_id = quota_project_id
```

Comment:
Most of the code in `__init__` above this line seems to be the same as that in the regular (non-pooled) `GrpcAsyncIOTransport`.  Does this suggest the possibility for the pool transport to be a subclass of the regular one?

Reply:
Yeah, I'm still considering how best to go about that. The regular GrpcAsyncIOTransport will attempt to create its own grpc channel and wrapped messages, which I'd like to avoid.

I think eventually we can try to remove the pooled transport entirely (see update in description)